app/code/Magento/Store/Controller/Store/Redirect.php

```diff
@@ -53,12 +60,14 @@ public function __construct(
         StoreResolverInterface $storeResolver,
         \Magento\Framework\Session\Generic $session,
         \Magento\Framework\Session\SidResolverInterface $sidResolver,
-        HashGenerator $hashGenerator
+        HashGenerator $hashGenerator,
+        StoreManagerInterface $storeManager
     ) {
         parent::__construct($context);
         $this->storeRepository = $storeRepository;
         $this->storeResolver = $storeResolver;
         $this->hashGenerator = $hashGenerator;
+        $this->storeManager = $storeManager;
```

Comment:
Please change to`$this->storeManager = $storeManager ?? ObjectManager::getInstance()->get(StoreManagerInterface::class);` for backward compatible
https://devdocs.magento.com/guides/v2.3/contributor-guide/backward-compatible-development/